src/librustdoc/html/static/js/search.js

```diff
@@ -320,215 +733,220 @@ window.initSearch = function(rawSearchIndex) {
                     path = result.item.path.toLowerCase(),
                     parent = result.item.parent;
 
-                if (!isType && !validateResult(name, path, split, parent)) {
+                if (!isType && !validateResult(name, path, nameSplit, parent)) {
                     result.id = -1;
                 }
             }
             return transformResults(results);
         }
 
-        function extractGenerics(val) {
-            val = val.toLowerCase();
-            if (val.indexOf("<") !== -1) {
-                var values = val.substring(val.indexOf("<") + 1, val.lastIndexOf(">"));
-                return {
-                    name: val.substring(0, val.indexOf("<")),
-                    generics: values.split(/\s*,\s*/),
-                };
+        /**
+         * This function checks if the object (`row`) generics match the given type (`elem`)
+         * generics. If there are no generics on `row`, `defaultLev` is returned.
+         *
+         * @param {Row} row            - The object to check.
+         * @param {QueryElement} elem  - The element from the parsed query.
+         * @param {integer} defaultLev - This is the value to return in case there are no generics.
+         *
+         * @return {integer}           - Returns the best match (if any) or `MAX_LEV_DISTANCE + 1`.
+         */
+        function checkGenerics(row, elem, defaultLev) {
```

Comment:
We have three very similar-sounding functions:

checkGenerics
checkIfInGenerics
checkType

As of right now, they all have more or less the same function comment. Can you explain how these functions differ, and then update the function comments to describe it?

Reply:
I'll use this function as an example for the explanations:

```rust
fn foo<T: Clone>(v: Vec<T>);
```

Let's start with `checkType`. If the user query is "Vec<Clone>", we will look into the functions' arguments and compare if any has a type named "Vec" with a generic (aka trait bound) named "Clone". So first, we check the type name itself ("Vec"). If the type name matches, then we want to checks its generics since our query has a generic. To do so, we call `checkGenerics` which will confirm that the "Vec" argument has a generic named "Clone".

Now, if the user query was just "Clone", then we won't match "Vec". But maybe "Clone" is in its generics, so we then call `CheckIfInGenerics` to check if it's the case.

Is it more clear like this? Maybe you have an idea on how to improve the documentation so it's easy to understand?